.github/workflows/ci.yml

```diff
@@ -4,6 +4,9 @@ on:
   workflow_dispatch:
   push:
   pull_request:
+    types:
+      - opened
+      - reopened
```

Comment:
If I pushed to a branch, waited for the build to finish, and then opened a PR, would it run the build again?

Reply:
I'd prefer the confusing approach, to be honest, because my normal workflow is that I push to a branch and only when the branch is passing do I open a pull request. There's always going to be an initial push to a branch that triggers a build (excluding the scenario where the PR is from a fork), so we're always going to have a duplicate run.

Reply:
Just sharing my workflow, I similarly create a PR after I see that my branch is passing.

Reply:
Just to clarify, CI will run when you push to your branch (with or without a PR). This optimization is to improve what happens after we open a PR.
So the only duplicate CI run will be once when we first open a PR. After that, CI will trigger for any pushes to your branch per usual.

Reply:
I guess what I'm saying is I don't want that duplicate CI run to happen when you open the PR.

 Because our tests have issues with flakiness, my general approach is to push the branch, make sure everything is passing so that I know it doesn't need either further changes because of a bug or retries because of intermittent failures, and then open the PR. Then at that point I can be done with this PR until it's reviewed and move on to another task.

If it runs again after the PR is opened, I'll have to check back again and make sure everything is still passing.

Reply:
Would it make sense to remove the `pull_request` block? I think with CircleCI, we typically ran tests only on push. I don't recall if any tests were run on pull requests opening / reopening. Would this solve the duplicate test runs? Or are there new actions that specifically need to be run on pull request?

Reply:
👍 @mkam, understood.
@lornasong, yes it's possible but we need CI to run when we get a PR from a forked repo. In that case, without the PR block, we won't have any PR checks.

Reply:
Thanks for that context. To confirm, it sounds like the only way to run tests on PRs from forked repos is to have a duplicate run when opening a PR against the original repo?

Reply:
@lornasong I just had a discussion with Melissa, sharing here for context. I have removed the pull_request trigger, this covers the majority of cases, CAT team creating a PR from a branch. I will test after merging this what happens when a PR is made from a forked repo and adjust as needed.

Reply:
> Thanks for that context. To confirm, it sounds like the only way to run tests on PRs from forked repos is to have a duplicate run when opening a PR against the original repo?

No, there's a way to only trigger on PRs from forked repos but it requires adding an `if` block to our workflow jobs. I wanted to see if we have a simpler solution first.